app/controllers/queue_items_controller.rb

```diff
@@ -41,15 +52,15 @@ def add_video_to_queue(video)
     end 
   end  
    
-  def reorder_queue_items
-    new_position = 1
-    @current_user.queue_items.each do |item|
-      item.queue_position = new_position
-      item.save
-      new_position += 1
-    end
+  def update_queue_items
+    ActiveRecord::Base.transaction do
+      params[:queue_items].each do |item|
+        queue_item = QueueItem.find(item["id"])
+        queue_item.update_attributes!(queue_position: item["queue_position"], rating: item["rating"]) if queue_item.user == current_user   
```

Comment:
This line lines is over 80 characters - to keep lines readable and to make debugging easier please refactor any lines over this limit. You can see this referenced in the style guide here - https://github.com/bbatsov/ruby-style-guide#80-character-limits